src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1546 @@
+     */
+    interface ArbitrarilyJumpableGenerator extends LeapableGenerator {
+
+        /**
+         * Returns an instance of {@link ArbitrarilyJumpableGenerator} that
+         * utilizes the {@code name} <a href="package-summary.html#algorithms">algorithm</a>.
+         *
+         * @param name  Name of random number generator
+         *              <a href="package-summary.html#algorithms">algorithm</a>
+         *
+         * @return An instance of {@link ArbitrarilyJumpableGenerator}
+         *
+         * @throws NullPointerException if name is null
+         * @throws IllegalArgumentException if the named algorithm is not found
+         */
+        static ArbitrarilyJumpableGenerator of(String name) {
+            Objects.requireNonNull(name);
+
+            return RandomGeneratorFactory.of(name, ArbitrarilyJumpableGenerator.class);
+        }
+
+        /**
+         * Returns a new generator whose internal state is an exact copy of this
+         * generator (therefore their future behavior should be identical if
+         * subjected to the same series of operations).
+         *
+         * @return a new object that is a copy of this generator
+         */
+        ArbitrarilyJumpableGenerator copy();
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a distance equal to 2<sup>{@code logDistance}</sup> within
+         * its state cycle.
+         *
+         * @param logDistance the base-2 logarithm of the distance to jump forward within the state
+         *                    cycle
+         *
+         * @throws IllegalArgumentException if {@code logDistance} is
+         *                                  2<sup>{@code logDistance}</sup> is
+         *                                  greater than the period of this generator
+         */
+        void jumpPowerOfTwo(int logDistance);
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a specified distance within its state cycle.
+         *
+         * @param distance the distance to jump forward within the state cycle
+         *
+         * @throws IllegalArgumentException if {@code distance} is not greater than
+         *                                  or equal to 0.0, or is greater than the
+         *                                  period of this generator
+         */
+        void jump(double distance);
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a large, fixed distance (typically 2<sup>64</sup> or more)
+         * within its state cycle. The distance used is that returned by method
+         * {@link ArbitrarilyJumpableGenerator#jumpDistance() jumpDistance}().
+         *
+         * @implSpec The default implementation invokes jump(jumpDistance()).
+         */
+        default void jump() { jump(jumpDistance()); }
```

Comment:
Should be able to avoid defining the jump method in this subinterface.

Reply:
jump(double distance) does not occur in superinterface. The ability to jump arbitrarily is specific to ArbitrarilyJumpableGenerator.